pyiron_base/jobs/job/core.py

```diff
@@ -429,6 +431,50 @@ def remove(self, _protect_childs=True):
         # After all children are deleted, remove the job itself.
         self.remove_child()
 
+    def kill(self, enable=True):
+        """
+        Kills a job via queueing system commands (e.g. scancel/qdel in terminal)
+        It updates the status of the job to "aborted" in the database.
+
+        enable (bool): Attempt a job-kill regardless of status when set to True.
+                       This will set the status to "aborted" no matter if the job was successfully killed or not!
+                       Set to False if you don't know what you are doing (unlikely).
+                       When False, it checks that the job is actually in a queued/running state
+                       in the database before attempting to remove it from the queue and set the aborted status.
+
+        Use case: A python-side equivalent method to call "scancel JOBID" (slurm) or "qdel JOBID" (PBSPro/torque) via pyiron
+        This command is "dumb" by default, it will attempt to kill anything that it is called on, regardless of job-status.
+        """
+
+        def base_kill():
+            if "server" in self.project_hdf5.list_nodes():
+                server_hdf_dict = self.project_hdf5["server"]
+                if (
+                    "qid" in server_hdf_dict.keys()
+                    and server_hdf_dict["qid"] is not None
+                ):
+                    self.project.queue_delete_job(server_hdf_dict["qid"])
+                    self._status = "aborted"
+                    self.project.db.item_update({"status": "aborted"}, self._job_id)
+            # TODO: implement a branch which enables killing of subprocesses if job._process exists
+            # see subprocess /pyiron_base/pyiron_base/jobs/job/runfunction.py --> run_job_with_runmode_non_modal
+            # PSEUDOCODE:
+            # elif hasattr(self, "_process"):
+            # INSERT KILL CODE FOR SUBPROCESS HERE
+
+        if enable:
+            base_kill()
+        else:
+            if self.status.running or self.status.submitted:
+                base_kill()
+            else:
+                raise ValueError(
+                    "The kill() function is only available during the execution of the job."
+                )
```

Comment:
Not sure I follow this logic.  If `enable` is given, we attempt to kill the job regardless of status and if not we only do it during the `running` or `submitted` status.  Given that `kill` only makes sense during those anyway and `enable` is meant only as footgun protection, I'd prefer more some like `if enable and (self.status.running or self.status.submitted)` and `enable=False` by default.

Reply:
I guess the point was that the status checking isn't foolproof. In an ideal world job statuses are rock-solid reliable and the force-kill option wouldn't be necessary, but there are some cases where this is not the case (e.g. Leimeroth's experience with the remote setup). Another example would be that the jobtable isn't updated with "aborted" when jobs hit walltime. 

There should be an option to kill the job regardless of the job status until we reach the point that funny business with job statuses is no longer the case.

Reply:
Oh, than I had always misunderstood this, my bad.  With this intention the 'force' or whatever we called it earlier is a better name.